projects/views.py

```diff
@@ -165,40 +166,46 @@ def view_project(request, project_id):
     project.rejected_allocations = []
     project.inactive_allocations = []
 
-    for a in project.allocations:
-        if a.status == 'Active' and a.resource == 'Chameleon':
+    project_allocations = Allocation.objects.filter(project_charge_code=project.chargeCode)
+
+    for a in project_allocations:
+        if a.status == 'Active':
             project.active_allocations.append(a)
             project.has_active_allocations = True
```

Comment:
General note on this: I think it would really clean up this code to make some of these "derived" properties actually derived from other state. For example, on the Project:

```python
@property
def has_active_allocations(self):
    return len(self.active_allocations) > 0

@property
def has_inactive_allocations(self):
    return len(project.inactive_allocations) > 0
```

Similarly, on your Allocation model:

```python
@property
def days_left(self):
    now = datetime.utcnow().replace(tzinfo=None)
    return (self.expiration_date.replace(tzinfo=None) - now).days

@property
def percent_su_used(self):
    if self.su_allocated and self.su_used:
        return self.su_used / self.su_allocated * 100
    else:
        return 0.0
```

This way, you only have to manage the "core" values of state, and the rest can be computed on-the-fly, as it "derives" from the core state naturally via some transformation, which is fixed.